src/transformers/training_args.py

```diff
@@ -1419,7 +1419,7 @@ def __post_init__(self):
             and (self.device.type != "cuda")
             and (self.device.type != "npu")
             and (self.device.type != "xpu")
-            and (get_xla_device_type(self.device) != "GPU")
+            and (get_xla_device_type(self.device) != "CUDA")
```

Comment:
Does this need to go into HF main?

Reply:
Yeah, this should. Let me create another PR in HF main.